### PR TITLE
feat: added `traits()` and `hasTraits()` functions

### LINKED ISSUE #226: No functions to access message and operation traits
#### Reason/Context
Even though we provide access to message and operation traits if listed in the components sections, there exist no functions on the operation and message objects themselves to access traits if you use the parser with the option `applyTraits = false`. 

#### Description
I would expect functions such as the following to be available:
Message trait: `asyncapi.channel('mychannel').publish().message().traits();`
Operation trait: `asyncapi.channel('mychannel').publish().traits();`

It is possible to access the traits using the following
Message trait: `asyncapi.channel('mychannel').publish().message()._json.traits`
Operation trait: `asyncapi.channel('mychannel').publish()._json.traits`

Up for discussion:
If the traits are already applied, would you really want access to the original traits 🤔 ?

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**

- this PR will add the above functions to the `Message` and `Operation` classes.

**Related issue(s)**
Resolves #226

### CODE REVIEW

**Comment by KhudaDad414** [comment]
This PR is ready for review.

**Review by magicmatatjahu — COMMENTED** [review]
@KhudaDad414 Thanks for contribution. If PR is ready, you can always request for review:

image

About PR, overall good, but I don't see logic for handling situation when someone pass `applyTraits: true` as option to the parser :) I wrote how it can be handled in this comment: https://github.com/asyncapi/parser-js/issues/226#issuecomment-914281717

File: test/models/message_test.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const { expect } = require('chai');
```
Comment (magicmatatjahu) [inline test/models/message_test.js:1]: Last thing. Could you add also test cases with `x-parser-original-traits` in Message and Operation object? :)

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

**Comment by KhudaDad414** [comment]
> @KhudaDad414 Thanks for contribution. If PR is ready, you can always request for review:
> 
@magicmatatjahu I don't have the option to request reviews. I can only re-request if someone has already reviewed the PR. 🤔 
Capture

**Comment by magicmatatjahu** [comment]
@KhudaDad414 It is also good, because then as previously reviewer I see your request again :)

**Review by magicmatatjahu — APPROVED** [review]
LGTM! Thanks for contribution!

**Comment by magicmatatjahu** [comment]
@all-contributors please add @KhudaDad414 for code, test

**Pull request merged**